src/librustc_trans/collector.rs

```diff
@@ -1019,7 +1019,9 @@ fn create_fn_trans_item<'a, 'tcx>(scx: &SharedCrateContext<'a, 'tcx>,
     let concrete_substs = monomorphize::apply_param_substs(scx,
                                                            param_substs,
                                                            &fn_substs);
-    assert!(concrete_substs.is_normalized_for_trans());
+    //assert!(concrete_substs.is_normalized_for_trans(),
+    //        "concrete_substs not normalized for trans: {:?}",
+    //        concrete_substs);
```

Comment:
Yes, this doesn't "loosen" the assertion, it disables it. Looks like the change to the assertion is incomplete or something.